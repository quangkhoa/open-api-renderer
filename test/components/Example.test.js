import React from 'react'
import Example from './../../src/components/Example/Example'
import renderer from 'react-test-renderer'

describe('<Example />', () => {
  it('renders nothing when there is no examples passed', () => {
    const tree = renderer.create(
      <Example examples={[]} />
    )

    expect(tree).toMatchSnapshot()
  })

  it('renders the summary only', () => {
    const tree = renderer.create(
      <Example examples={[{'summary': 'This is a test summary.'}]} />
    )

    expect(tree).toMatchSnapshot()
  })

  it('renders the description only', () => {
    const tree = renderer.create(
      <Example examples={[{'description': 'This is a test description.'}]} />
    )

    expect(tree).toMatchSnapshot()
  })

  it('renders a string example', () => {
    let tree = renderer.create(
      <Example examples={[{'value': '<message>hi</message>'}]} />
    )

    expect(tree).toMatchSnapshot()
  })

  it('renders a JSON example', () => {
    const tree = renderer.create(
      <Example examples={[{'value': {'message': 'hi'}}]} />
    )

    expect(tree).toMatchSnapshot()
  })

  it('renders a string example with a summary', () => {
    let tree = renderer.create(
      <Example examples={[{'summary': 'This is a test summary.', 'value': '<message>hi</message>'}]} />
    )

    expect(tree).toMatchSnapshot()
  })

  it('renders a JSON example with a summary', () => {
    const tree = renderer.create(
      <Example examples={[{'summary': 'This is a test summary.', 'value': {'message': 'hi'}}]} />
    )

    expect(tree).toMatchSnapshot()
  })

  it('renders a string example with a description', () => {
    let tree = renderer.create(
      <Example examples={[{'description': 'This is a test description.', 'value': '<message>hi</message>'}]} />
    )

    expect(tree).toMatchSnapshot()
  })

  it('renders a JSON example with a description', () => {
    const tree = renderer.create(
      <Example examples={[{'description': 'This is a test description.', 'value': {'message': 'hi'}}]} />
    )

    expect(tree).toMatchSnapshot()
  })

  it('renders a string example with a summary and a description', () => {
    let tree = renderer.create(
      <Example examples={[{'summary': 'This is a test summary.', 'description': 'This is a test description.', 'value': '<message>hi</message>'}]} />
    )

    expect(tree).toMatchSnapshot()
  })

  it('renders a JSON example with a summary and a description', () => {
    const tree = renderer.create(
      <Example examples={[{'summary': 'This is a test summary.', 'description': 'This is a test description.', 'value': {'message': 'hi'}}]} />
    )

    expect(tree).toMatchSnapshot()
  })
})
