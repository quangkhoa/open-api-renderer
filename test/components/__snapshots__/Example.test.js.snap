// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Example /> renders a JSON example with a summary and a description 1`] = `
<div
  className={undefined}
>
  <div
    className={undefined}
  >
    This is a test summary.
  </div>
  <div
    className={undefined}
  >
    This is a test description.
  </div>
  <div
    className={undefined}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className=""
    >
      <span
        onClick={[Function]}
        onMouseOver={[Function]}
        title="Copy to Clipboard"
      >
        Copy
      </span>
    </div>
    <pre>
      {
  "message": "hi"
}
    </pre>
  </div>
</div>
`;

exports[`<Example /> renders a string example with a summary and a description 1`] = `
<div
  className={undefined}
>
  <div
    className={undefined}
  >
    This is a test summary.
  </div>
  <div
    className={undefined}
  >
    This is a test description.
  </div>
  <div
    className={undefined}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className=""
    >
      <span
        onClick={[Function]}
        onMouseOver={[Function]}
        title="Copy to Clipboard"
      >
        Copy
      </span>
      <span
        onClick={[Function]}
      >
        Expand All
      </span>
      <span
        onClick={[Function]}
      >
        Collapse All
      </span>
    </div>
    <pre>
      &lt;message&gt;hi&lt;/message&gt;
    </pre>
  </div>
</div>
`;

exports[`<Example /> renders a JSON example with a description 1`] = `
<div
  className={undefined}
>
  <div
    className={undefined}
  >
    This is a test description.
  </div>
  <div
    className={undefined}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className=""
    >
      <span
        onClick={[Function]}
        onMouseOver={[Function]}
        title="Copy to Clipboard"
      >
        Copy
      </span>
    </div>
    <pre>
      {
  "message": "hi"
}
    </pre>
  </div>
</div>
`;

exports[`<Example /> renders a string example with a description 1`] = `
<div
  className={undefined}
>
  <div
    className={undefined}
  >
    This is a test description.
  </div>
  <div
    className={undefined}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className=""
    >
      <span
        onClick={[Function]}
        onMouseOver={[Function]}
        title="Copy to Clipboard"
      >
        Copy
      </span>
      <span
        onClick={[Function]}
      >
        Expand All
      </span>
      <span
        onClick={[Function]}
      >
        Collapse All
      </span>
    </div>
    <pre>
      &lt;message&gt;hi&lt;/message&gt;
    </pre>
  </div>
</div>
`;

exports[`<Example /> renders a JSON example with a summary 1`] = `
<div
  className={undefined}
>
  <div
    className={undefined}
  >
    This is a test summary.
  </div>
  <div
    className={undefined}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className=""
    >
      <span
        onClick={[Function]}
        onMouseOver={[Function]}
        title="Copy to Clipboard"
      >
        Copy
      </span>
    </div>
    <pre>
      {
  "message": "hi"
}
    </pre>
  </div>
</div>
`;

exports[`<Example /> renders a string example with a summary 1`] = `
<div
  className={undefined}
>
  <div
    className={undefined}
  >
    This is a test summary.
  </div>
  <div
    className={undefined}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className=""
    >
      <span
        onClick={[Function]}
        onMouseOver={[Function]}
        title="Copy to Clipboard"
      >
        Copy
      </span>
      <span
        onClick={[Function]}
      >
        Expand All
      </span>
      <span
        onClick={[Function]}
      >
        Collapse All
      </span>
    </div>
    <pre>
      &lt;message&gt;hi&lt;/message&gt;
    </pre>
  </div>
</div>
`;

exports[`<Example /> renders a JSON example 1`] = `
<div
  className={undefined}
>
  <div
    className={undefined}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className=""
    >
      <span
        onClick={[Function]}
        onMouseOver={[Function]}
        title="Copy to Clipboard"
      >
        Copy
      </span>
    </div>
    <pre>
      {
  "message": "hi"
}
    </pre>
  </div>
</div>
`;

exports[`<Example /> renders a string example 1`] = `
<div
  className={undefined}
>
  <div
    className={undefined}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className=""
    >
      <span
        onClick={[Function]}
        onMouseOver={[Function]}
        title="Copy to Clipboard"
      >
        Copy
      </span>
      <span
        onClick={[Function]}
      >
        Expand All
      </span>
      <span
        onClick={[Function]}
      >
        Collapse All
      </span>
    </div>
    <pre>
      &lt;message&gt;hi&lt;/message&gt;
    </pre>
  </div>
</div>
`;

exports[`<Example /> renders the description only 1`] = `
<div
  className={undefined}
>
  <div
    className={undefined}
  >
    This is a test description.
  </div>
</div>
`;

exports[`<Example /> renders the summary only 1`] = `
<div
  className={undefined}
>
  <div
    className={undefined}
  >
    This is a test summary.
  </div>
</div>
`;

exports[`<Example /> renders nothing when there is no examples passed 1`] = `null`;
